parsers/Parser.js

```diff
@@ -7,31 +9,97 @@ module.exports = class Parser {
         this.document = document;
     }
 
-    parse() {
-        return [];
+    getRules() {
```

Comment:
With the addition of the `get rules()` getter, this becomes confusing.  It isn't clear when it is appropriate to use `parser.getRules()` vs `parser.rules`.